allensdk/brain_observatory/behavior/data_objects/running_speed/running_speed.py

```diff
@@ -0,0 +1,179 @@
+
+import json
+from typing import Optional
+
+from cachetools import cached, LRUCache
+from cachetools.keys import hashkey
+
+import pandas as pd
+
+from pynwb import NWBFile, ProcessingModule
+from pynwb.base import TimeSeries
+
+from allensdk.core.exceptions import DataFrameIndexError
+from allensdk.internal.api import PostgresQueryMixin
+from allensdk.brain_observatory.behavior.data_objects import (
+    DataObject, StimulusTimestamps
+)
+from allensdk.brain_observatory.behavior.data_files import (
+    StimulusFile
+)
+from allensdk.brain_observatory.behavior.data_objects.running_speed.running_processing import (  # noqa: E501
+    get_running_df
+)
+
+
+def from_json_cache_key(
+    cls, dict_repr: dict, filtered: bool, zscore_threshold: float
+):
+    return hashkey(json.dumps(dict_repr), filtered, zscore_threshold)
+
+
+def from_lims_cache_key(
+    cls, db, behavior_session_id: int, ophys_experiment_id: Optional[int],
+    filtered: bool, zscore_threshold: float
+):
+    return hashkey(
+        behavior_session_id, ophys_experiment_id, filtered, zscore_threshold
+    )
+
+
+class RunningSpeed(DataObject):
+    def __init__(
+        self,
+        running_speed: pd.DataFrame,
+        stimulus_file: Optional[StimulusFile] = None,
+        stimulus_timestamps: Optional[StimulusTimestamps] = None,
+        filtered: bool = True
+    ):
+        super().__init__(name='running_speed', value=running_speed)
+        self._stimulus_file = stimulus_file
+        self._stimulus_timestamps = stimulus_timestamps
+        self._filtered = filtered
+
+    @staticmethod
+    def _get_running_speed_df(
+        stimulus_file: StimulusFile,
+        stimulus_timestamps: StimulusTimestamps,
+        filtered: bool = True,
+        zscore_threshold: float = 1.0
+    ) -> pd.DataFrame:
+        running_data_df = get_running_df(
```

Comment:
why does `get_running_df` not live inside of a `DataObject` ?

Reply:
Both `RunningAcquisition` and `RunningSpeed` DataObjects use `get_running_df` I decided that a reasonable compromise would be to move `allensdk/brain_observatory/behavior/running_processing.py` to `allensdk/brain_observatory/behavior/data_objects/running_speed/running_processing.py`.